packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	tokens,
+} from "@fluentui/react-components";
+import { DataEdit, FluidObjectValueNode, Primitive } from "@fluid-experimental/devtools-core";
+
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableValueView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link ContainerFeaturesContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	/**
+	 * State to store whether the user is currently editing data
+	 */
+	const [editing, setEditing] = React.useState<boolean>(false);
+
+	/**
+	 * State to store the current type, or the type of the intended edit
+	 */
+	const [type, setType] = React.useState<string>(typeof node.value);
+
+	/**
+	 * State to store the corresponding UI component
+	 */
+	const [editingComponent, setEditingComponent] = React.useState<React.ReactElement>(<></>);
+
+	const messageRelay = useMessageRelay();
+
+	/**
+	 * On blur will set {@link editing} to false
+	 */
+	const onBlur = React.useCallback((event: React.FocusEvent<HTMLDivElement>) => {
+		/**
+		 * This allows the parent element to share focus with children
+		 * so the blur event only triggers if focus is outside of div and children
+		 */
+		if (!event.currentTarget.contains(event.relatedTarget)) {
+			setEditing(false);
+		}
+	}, []);
```

Comment:
Since this callback doesn't have any dependencies, I don't think we need to use a react callback - this could just be a standard function.

Reply:
Should we just in-line these or keep them seperate in case more functionality wnats to be added to them?